hazelcast/src/main/java/com/hazelcast/internal/cluster/impl/MembershipManager.java

```diff
@@ -803,7 +803,7 @@ void sendMembershipEvents(Collection<MemberImpl> currentMembers, Collection<Memb
                 node.getPartitionService().memberAdded(newMember);
                 node.getNodeExtension().onMemberListChange();
                 Joiner joiner = node.getJoiner();
-                if (joiner != null && joiner.getClass() == TcpIpJoiner.class) {
+                if (joiner instanceof TcpIpJoiner) {
```

Comment:
Same as above, we should revert this.

Reply:
https://github.com/hazelcast/hazelcast/pull/21552/commits/7c02c23141de4d44cf3f219dd9b0497c47def776